src/App.js

```diff
@@ -0,0 +1,297 @@
+import './App.css';
+import React from 'react';
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            products: [
+                {id: '1', name: 'Apple', price: 2},
+                {id: '2', name: 'Banana', price: 1},
+                {id: '3', name: 'Chokeberry', price: 5},
+                {id: '4', name: 'Raspberry', price: 4},
+                {id: '5', name: 'Strawberry', price: 3},
+                {id: '6', name: 'Blackberry', price: 4},
+                {id: '7', name: 'Mango', price: 6},
+                {id: '8', name: 'Orange', price: 2},
+                {id: '9', name: 'Kiwi', price: 1},
+                {id: '10', name: 'Lemon', price: 2},
+                {id: '11', name: 'Melon', price: 3},
+                {id: '12', name: 'Peach', price: 2},
+                {id: '13', name: 'Fig tree', price: 3},
+                {id: '14', name: 'Pear', price: 2},
+                {id: '15', name: 'Blueberry', price: 5},
+            ],
+            allProducts: [
+                {id: '1', name: 'Apple', price: 2},
+                {id: '2', name: 'Banana', price: 1},
+                {id: '3', name: 'Chokeberry', price: 5},
+                {id: '4', name: 'Raspberry', price: 4},
+                {id: '5', name: 'Strawberry', price: 3},
+                {id: '6', name: 'Blackberry', price: 4},
+                {id: '7', name: 'Mango', price: 6},
+                {id: '8', name: 'Orange', price: 2},
+                {id: '9', name: 'Kiwi', price: 1},
+                {id: '10', name: 'Lemon', price: 2},
+                {id: '11', name: 'Melon', price: 3},
+                {id: '12', name: 'Peach', price: 2},
+                {id: '13', name: 'Fig tree', price: 3},
+                {id: '14', name: 'Pear', price: 2},
+                {id: '15', name: 'Blueberry', price: 5},
+            ],
+            cart: JSON.parse(localStorage.getItem('sessionCartValue')) || [],
+            total: 0,
+            searchText: '',
+        };
+
+        let sum = 0;
+        for (let i = 0; i < this.state.cart.length; i++) {
+            sum += this.state.cart[i].count * this.state.cart[i].price;
+        }
+        this.state.total = sum;
+
+        this.addToCart = this.addToCart.bind(this);
+        this.remove = this.remove.bind(this);
+        this.decreaseCount = this.decreaseCount.bind(this);
+        this.increaseCount = this.increaseCount.bind(this);
+        this.removeAll = this.removeAll.bind(this);
+        this.search = this.search.bind(this);
+    }
+
+    addToCart(product) {
+        let tmpCart = this.state.cart;
```

Comment:
Ne treba `let` ako nemas redekleraciju variable (npr. `tmpCart = ...`)
```suggestion
        const tmpCart = this.state.cart;
```